src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -10,7 +10,7 @@
 
 namespace Microsoft.CodeAnalysis.Operations
 {
-    internal sealed class ControlFlowGraphBuilder : OperationCloner
+    internal sealed class ControlFlowGraphBuilder : OperationVisitor<int?, IOperation>
```

Comment:
Perhaps we should consider creating a version of the cloner that passes and argument, and have the default cloner pass null? I hate to have all the duplicated new calls below.

Reply:
> Perhaps we should consider creating a version of the cloner that passes an argument, and have the default cloner pass null? I hate to have all the duplicated new calls below.

They are not duplicated, they all already are different in the way they handle SemanticModel and pretty much all of them are going to diverge even more as we add proper handling for control flow in those nodes (stack spilling, etc.). We are just starting.

---
In reply to: [163338241](https://github.com/dotnet/roslyn/pull/24358#discussion_r163338241) [](ancestors = 163338241)